src/bin/julialauncher.rs

```diff
@@ -344,19 +531,23 @@ fn run_app() -> Result<i32> {
             ));
         };
 
-    let (julia_path, julia_args) = get_julia_path_from_channel(
-        &versiondb_data,
-        &config_file.data,
-        &julia_channel_to_use,
-        &paths.juliaupconfig,
-        juliaup_channel_source,
-    )
-    .with_context(|| {
-        format!(
-            "The Julia launcher failed to determine the command for the `{}` channel.",
-            julia_channel_to_use
+    let (julia_path, julia_args) = if let Some(path) = manifest_derived_julia_path {
```

Comment:
We probably should think the priority list here through a bit more... So with this PR here it is
1. Command line switch
2. Manifest
3. `JULIAUP_CHANNEL` env var
4. Folder override
5. Default

That might be the right choice, but we should just make sure that we think it is :)

Reply:
How should it handle a situation where an environment has multiple versioned manifests?

For example, in my Metal.jl dev environment, I have `Manifest.toml`, `Manifest-v1.11.toml`, and `Manifest-v1.12.toml`.

Maybe if there's a manifest that matches the next-highest priority setting, select that?

For example, if I start Julia with 
```bash
$ JULIAUP_CHANNEL=rc julia --project=.
```
in an environment where my default channel is `release`, and that I have `Manifest.toml`, `Manifest-v1.11.toml`, and `Manifest-v1.12.toml` in that folder, it would use `Manifest-v1.11.toml` since the next priority is `JULIAUP_CHANNEL`, and it's set to `rc`.

Calling 
```bash
$ julia --project=.
```
would launch v1.10 because that's the default channel.

I'm not exactly sure this should interact with the Julia Manifest searching priority order (listed in my other review comment), but it's worth figuring out.

Reply:
Responding to my own post here, I think the order should be this:
1. Command line switch
2. `JULIAUP_CHANNEL` env var
3. Folder override
4. Manifest
5. Default

@christiangnrd I think it essentially doesn't, i.e. the Juliaup launcher will just always completely ignore versioned manifest files. If there is a `Manifest.toml`, then it will (by default) launch the Julia version in there, and then that Julia version will use that manifest. If one launches with say a command line flag `+1.11`, _and_ there is a 1.11 specific manifest, then Julia itself will of course use that manifest, but the code here in Juliaup doesn't know about this, nor does it need.

At some level this would all be much easier if one could just select a manifest file by name (and they could be freely named)... I think the ideal design would be `julia +manifestfilename` really... Say we had `julia +mymanifest.juliamanifest.toml` convention or something like that...But that would require changes in Julia base, and I'm not sure whether one could come up with a good structure there.

Reply:
This is now the order used.